crates/rattler_solve/src/resolvo/mod.rs

```diff
@@ -555,7 +714,9 @@ impl<'a> DependencyProvider for CondaDependencyProvider<'a> {
                         return Dependencies::Unknown(reason);
                     }
                 };
-            dependencies.constrains.push(version_set_id);
+            for version_set_id in version_set_id {
+                dependencies.constrains.push(version_set_id);
+            }
```

Comment:
Might be better to use `extend` to ensure we dont push individual items and allocate multiple times.